sdk/samples/iot/provisioning/src/paho_iot_provisioning_example.c

```diff
@@ -491,16 +491,11 @@ static void sleep_for_seconds(uint32_t seconds)
 }
 
 // Print an az_span to the console
-static void print_az_span(const char* str, az_span span)
+static int printf_az_span(const char* str, az_span span)
 {
-  printf("%s", str);
+  char buffer[az_span_size(span) + 1];
+  memcpy(buffer, az_span_ptr(span), (uint32_t)az_span_size(span));
+  buffer[az_span_size(span)] = '\0';
```

Comment:
This helper shouldn't be necessary and is redundant.

I would consider using the existing, provided API, which essentially does what you are trying to do here:
`az_span_to_str(...)`

https://github.com/Azure/azure-sdk-for-c/blob/eb836ef1af71515b5644f40cb2a163389a250801/sdk/inc/azure/core/az_span.h#L238

One reason to avoid your own custom implementations is that use of `memcpy` isn't resilient to overlapping buffers, which is why `az_span_to_str` using `memmove`.

Reply:
I just found az_span_to_str haha, but didn't know about it when I wrote this :)  Actually, I talked to Himanshu and he explained to me that because some az_spans could include multi-byte characters, we want to print out bytes one by one.  I am going to revert to the previous putchar approach Cris had.  The helper helps in that case.

Reply:
Does putchar support printing out multi-byte characters? Aren't we only giving it one byte anyway so how is it's behavior different from printf?

Reply:
For example, suppose we have a chinese character string: 汉
The is 3 bytes in UTF-8: 0xE6 0xB1 0x89, so imagine you have an az_span of size 3 containing those bytes.
https://www.fileformat.info/info/unicode/char/6c49/index.htm

What does printf output when turning that span to a string versus calling putchar 3 times? I would have thought both end up with the same output.

Reply:
If I recall correctly, @hihigupt explained that printf can drop bytes, such as if there are multiple 0s next to each other.  Himanshu, can you please clarify for me?  Thanks.

Reply:
@ahsonkhan  Reason we have putchar API is if span constrains binary data.. and we use printf then data might be truncated .... so to print everything using putchar.